modules/phase_field/src/kernels/ACGrGrPolyStoredEnergy.C

```diff
@@ -0,0 +1,58 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "ACGrGrPolyStoredEnergy.h"
+
+template<>
+InputParameters validParams<ACGrGrPolyStoredEnergy>()
+{
+  InputParameters params = validParams<ACBulk>();
+  params.addClassDescription("Stored energy contribution for the poly crystaline Allen-Cahn Kernel");
+  params.addRequiredParam<std::vector<Real> >("stored_energy", "list of stored energies for each grain in the simulation");
```

Comment:
I'm open for other suggestions. I could add a stored energy provider, like the EulerAngleProvider. That would allow users to use a custom column in an EBSD file to specify the stored energy.